### PR TITLE
[match] Fix not respecting git user options when nuking or changing password

### PR DESCRIPTION
**Pull request by KrauseFx**
- Also added comments to not forget it in the future
- Also added respecting of `clone_branch_directly` for other calls too besides runner.rb

Fixes https://github.com/fastlane/fastlane/issues/9097

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
Match::CommandsGenerator :decrypt option handling can use the git_url short flag from tool options | <a href='https://github.com/fastlane/fastlane/blob/389dec5924e3849ca0f1d221d59edd740cd92116/./match/spec/commands_generator_spec.rb'>commands_generator_spec.rb</a>|
Match::CommandsGenerator :decrypt option handling can use the shallow_clone flag from tool options | <a href='https://github.com/fastlane/fastlane/blob/389dec5924e3849ca0f1d221d59edd740cd92116/./match/spec/commands_generator_spec.rb'>commands_generator_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: match/lib/match/git_helper.rb [hunk 1]
```diff
@@ -1,6 +1,19 @@
 module Match
   class GitHelper
-    def self.clone(git_url, shallow_clone, manual_password: nil, skip_docs: false, branch: "master", git_full_name: nil, git_user_email: nil, clone_branch_directly: false)
+    def self.clone(git_url,
+                   shallow_clone,
```
Comment (ohayon) [inline match/lib/match/git_helper.rb]: Should this maybe have a named parameter?

File: match/lib/match/git_helper.rb [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+                   manual_password: nil,
+                   skip_docs: false,
+                   branch: "master",
+                   git_full_name: nil,
+                   git_user_email: nil,
+                   clone_branch_directly: false)
```
Comment (ohayon) [inline match/lib/match/git_helper.rb]: What does this mean?

File: match/lib/match/git_helper.rb (hunk 1 above)
Comment (ohayon) [inline match/lib/match/git_helper.rb]: Talked about this off line, going to make this change in a separate PR.

File: match/lib/match/git_helper.rb (hunk 2 above)
Comment (ohayon) [inline match/lib/match/git_helper.rb]: @KrauseFx pointed me at this option. 👍

**Review by ohayon — APPROVED** [review]
👍
